litmus-agent-subscriber/src/main.go

```diff
@@ -0,0 +1,324 @@
+			utils.Red.Println("Error in getting agent list: ", err)
+			os.Exit(1)
+		}
+		return agentResp
+	}
+	return agentResponse{}
+}
+
+func GetAgentWithID(credentials types.Credentials, cluster_id string) agentData {
+	agents := GetAgentList(credentials, LITMUS_PROJECT_ID)
+	for _, agent := range agents.Data.GetCluster {
+		if agent.ClusterID == cluster_id {
+			return agent
+		}
+	}
+	return agentData{}
+
+}
+
+func GetAgentWithName(credentials types.Credentials, AgentName string) agentData {
+	agents := GetAgentList(credentials, LITMUS_PROJECT_ID)
+	for _, agent := range agents.Data.GetCluster {
+		if agent.ClusterName == AgentName {
+			return agent
+		}
+	}
+	return agentData{}
+}
+
+func createConfigMap(configmapName string, configMapData map[string]string) {
+	var err error
+	config, err := rest.InClusterConfig()
+	if err != nil {
+                utils.Red.Println("\n❌ Cannot create config from incluster: " + err.Error() + "\n")
+	}
+	// creates the clientset
+	clientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+                utils.Red.Println("\n❌ Cannot create clientset: " + err.Error() + "\n")
+	}
+
+	configMap := corev1r.ConfigMap{
+		TypeMeta: metav1r.TypeMeta{
+			Kind:       "ConfigMap",
+			APIVersion: "v1",
+		},
+		ObjectMeta: metav1r.ObjectMeta{
+			Name:      configmapName,
+			Namespace: NAMESPACE,
+		},
+		Data: configMapData,
+	}
+
+	var cm *corev1r.ConfigMap
+	if _, err := clientset.CoreV1().ConfigMaps(NAMESPACE).Get(context.TODO(), configmapName, metav1r.GetOptions{}); errors.IsNotFound(err) {
+		cm, err = clientset.CoreV1().ConfigMaps(NAMESPACE).Create(context.TODO(), &configMap, metav1r.CreateOptions{})
+		if err != nil {
+			utils.Red.Println("\n❌ Cannot create configmap " + configmapName + " : " + err.Error() + "\n")
+			os.Exit(1)
+		}
+
+	} else {
+		cm, err = clientset.CoreV1().ConfigMaps(NAMESPACE).Update(context.TODO(), &configMap, metav1r.UpdateOptions{})
+		if err != nil {
+			utils.Red.Println("\n❌ Cannot update configmap " + configmapName + " : " + err.Error() + "\n")
+			os.Exit(1)
+		}
+
+	}
+	_ = cm
+}
+
+func createAgent(credentials types.Credentials) {
+	var newAgent types.Agent
+	newAgent.AgentName = AGENT_NAME
+	newAgent.Namespace = NAMESPACE
+	newAgent.Description = AGENT_DESCRIPTION
+	newAgent.ProjectId = LITMUS_PROJECT_ID
+	// Mode
+	// 1. cluster
+	// 2. namespace
+	newAgent.Mode = "cluster"
+
+	// -- OPTIONNAL -- //
+	newAgent.ClusterType = "external"
+	newAgent.NodeSelector = ""
+	// PlatformName
+	// 1. AWS
+	// 2. GKE
+	// 3. Openshift
+	// 4. Rancher
+	// 5. Others
+	newAgent.PlatformName = "Others"
+	newAgent.ServiceAccount = SERVICE_ACCOUNT_NAME
+	newAgent.SAExists = true
+	newAgent.NsExists = true
+
+	t := time.Now()
+	configMapData := make(map[string]string)
+	configMapData["SERVER_ADDR"] = LITMUS_BACKEND_URL + "/query"
+	configMapData["VERSION"] = APP_VERSION
+	configMapData["IS_CLUSTER_CONFIRMED"] = "false"
+	configMapData["START_TIME"] = t.Format("20060102150405")
```

Comment:
Should be the current time time.Now().Unix()

Reply:
Hahaha xD
I found this dirty method to get unix format :D I did not know the .Unix() function of time.Now()
I did like -> time.Now().Format("20060102150405")
I get the current time line 199.

Your method is way better :)